### PR TITLE
feat: add MaxKeys typed parameter with validation and ergonomic API

### PR DESCRIPTION
**Pull request by HJLebbink**
feat: add MaxKeys typed parameter with validation and ergonomic API

  Implements client-side validation for the max_keys parameter used in
  ListObjects API calls, replacing implicit panics with explicit ValidationErr
  handling. This ensures invalid user input is caught early with clear error
  messages rather than causing runtime panics.

  Key improvements:

  - Adds MaxKeys typed parameter with validation (range: 1-1000) that matches
    MinIO EOS hard limits
  - Provides ergonomic builder API: .max_keys(500) instead of requiring
    .max_keys(MaxKeys::new(500)?)
  - Defers validation to request building time via to_s3request(), allowing
    graceful error propagation instead of panicking in From<u16>
  - Updates client documentation to reflect client-side validation behavior
  - Adds 17 comprehensive unit tests covering boundary conditions, parsing,
    and conversions
  - Fixes error message redundancy to match project patterns
  - Relaxes integration test assertions to respect max_keys as an upper bound,
    preventing flakiness across S3 implementations
  - Removes dead code (unused MAX_KEYS_TESTING.md, new_unchecked())

  The builder now accepts max_keys values via multiple ergonomic paths:
  - .max_keys(500) - accepts u16 directly
  - .max_keys(MaxKeys::new(500)?) - explicit pre-validation
  - .max_keys(None) - use server default (1000)

  Validation ensures compliance with MinIO/S3 constraints while maintaining
  clean error handling throughout the request lifecycle.

### CODE REVIEW

File: src/s3/types/typed_parameters.rs [hunk 1]
```diff
@@ -907,6 +907,143 @@ impl From<&ContentType> for ContentType {
...
+    }
+}
+
+impl std::str::FromStr for MaxKeys {
+    type Err = ValidationErr;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let value = s
+            .parse::<u16>()
+            .map_err(|_| ValidationErr::InvalidMaxKeys(format!("not a valid number: {}", s)))?;
```
[the remark this answers is not in the document]
Comment (HJLebbink) [reply src/s3/types/typed_parameters.rs:998]: what? nothing better todo?
